src/iexamine.cpp

```diff
@@ -4559,6 +4562,21 @@ void iexamine::autodoc( player &p, const tripoint &examp )
             }
             break;
         }
+        case TYMPANOPLASTY: {
+            if( !patient.has_effect( effect_deaf ) ) {
```

Comment:
There should also be a check for duration of effect being more than 2 days. Otherwise using this procedure will only extend the effect.